Parser/tokenizer.c

```diff
@@ -2448,27 +2446,28 @@ tok_get_fstring_mode(struct tok_state *tok, tokenizer_mode* current_tok, struct
         return tok_get_normal_mode(tok, current_tok, token);
     }
 
-    // Emit FSTRING_END in case we've reached the end of the string
-    if (start_char == current_tok->f_string_quote
-        && (current_tok->f_string_quote_size != 3 || (start_char == peek1 && start_char == peek2))) {
-        // Advance the tokenizer state again to create a token out of the end quotes
-        for (int i = 0; i < current_tok->f_string_quote_size; i++) {
-            tok_nextc(tok);
-        }
-
-        if (current_tok->last_expr_buffer != NULL) {
-            PyMem_Free(current_tok->last_expr_buffer);
-            current_tok->last_expr_buffer = NULL;
-            current_tok->last_expr_size = 0;
-            current_tok->last_expr_end = -1;
+    // Check if we are at the end of the string
+    for (int i = 0; i < current_tok->f_string_quote_size; i++) {
```

Comment:
The main difference is it doesn't peek unnecessarily which makes it not move the buffer to the next line which would make it impossible for tokenizer to know the line for the original start characters.